aiohttp/web_response.py

```diff
@@ -409,35 +455,38 @@ def _start_compression(self, request):
         self._body_length = self._payload_writer.output_size
         self._payload_writer = None
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         if self._eof_sent:
             info = "eof"
         elif self.prepared:
-            info = "{} {} ".format(self._req.method, self._req.path)
+            info = "{} {} ".format(
+                getattr(self._req, 'method'),
+                getattr(self._req, 'path'),
+            )
         else:
             info = "not prepared"
         return "<{} {} {}>".format(self.__class__.__name__,
                                    self.reason, info)
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: str) -> str:
         return self._state[key]
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key: str, value: str) -> None:
         self._state[key] = value
 
-    def __delitem__(self, key):
+    def __delitem__(self, key: str) -> None:
         del self._state[key]
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._state)
 
-    def __iter__(self):
-        return iter(self._state)
+    def __iter__(self) -> Iterator[Tuple[str, str]]:
```

Comment:
I will try to look into this once again  but basically this one says that `aiohttp/web_response.py:483: error: Return type of "__iter__" incompatible with supertype "Iterable"
`. An `Iterable` is last or the one before last in the chain of inheritance starting from `MutableMapping`. I can't really say why this one is bad if typings for `MutableMapping` says exactly `MutableMapping[str, str]`.